node/src/reactor/joiner.rs

```diff
@@ -909,7 +929,12 @@ impl Reactor {
     /// Deconstructs the reactor into config useful for creating a Validator reactor. Shuts down
     /// the network, closing all incoming and outgoing connections, and frees up the listening
     /// socket.
-    pub async fn into_validator_config(self) -> ValidatorInitConfig {
+    pub async fn into_validator_config(self) -> Result<ValidatorInitConfig, Error> {
+        // Clean the state of the linear_chain_sync before shutting it down.
+        linear_chain_sync::clean_linear_chain_state(
+            &self.storage,
+            self.chainspec_loader.chainspec(),
+        )?;
```

Comment:
```suggestion
        #[cfg(not(feature = "fast-sync"))]
        linear_chain_sync::clean_linear_chain_state(
            &self.storage,
            self.chainspec_loader.chainspec(),
        )?;
```

Reply:
This feels dirty but I will do that :)

Reply:
Done in [`087dddd` (#904)](https://github.com/CasperLabs/casper-node/pull/904/commits/087dddd78b6b23012eca2d7f765adf874623c248).